In [1]:
import math

__1. Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.__ 



Дано:<br>
$\sigma = 16$<br>
$M(x) = a$<br>
$p = 0.95$<br>
$Xв. = 80$<br>
$n = 256$<br>


1) Уже были выполнен анализ ГС известно $\sigma$, значит критерий оценки $K$ является <font color = red>критерием Фишера</font> $Z$ из таблицы распределения <font color = red>Лапласа</font><br>
2) Надежность 0.95 => $p = 0.95$<br>
$\alpha = 1 - p => \alpha = 0.05$<br>
3) ГС распределена нормально => можем от $\alpha$ перейти к $\frac{\alpha} { 2}$<br>
$\alpha / 2 = 0.05 / 2 = 0.025$<br>
4) Из 1) => $Ст.ош.ср. = \sigma / \sqrt{n}$<br>
$[X_{в.} - Z_{\alpha/2} \cdot \dfrac{\sigma}{\sqrt{n}};X_{в.} + Z_{\alpha/2} \cdot \dfrac{\sigma}{\sqrt{n}}]$<br>
$Z(p=0.025)=Z(\alpha /2=0.025)=-1.96$<br>


In [2]:
a=math.sqrt(256)
print(f'[80 - 1.96(sigma/sqrt(256)); 80 - 1.96(sigma/sqrt(256))] = [{80. - 1.96 * (16. / a)}; {80. + 1.96 * (16. / a)}]')

[80 - 1.96(sigma/sqrt(256)); 80 - 1.96(sigma/sqrt(256))] = [78.04; 81.96]


Вывод: $[78.04; 81.96]$ - этот интервал с вероятностью 95% покрывает истенное $M(x)=a$ генеральной совокупности.

__2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:<br>
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1<br>
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.__

$p=0.95 => \alpha = 1-p = 0.05$<br>
$n=10$<br>
$[Xв - K \cdot Ст.ош.ср; Xв - K \cdot Ст.ош.ср]$<br>
1) Неизвестна $\sigma$ => небыло анализа ГС.<br>
2) Из пункта 1) => критерий оценки K это критерий <font color = red>Стьюдента t</font><br>
$K=n-1 => K=10-1=9$<br>
3) Из 1) => $Ст.ош.ср. = \dfrac{s}{\sqrt{n}}$, где<br>
$s = \sqrt{D(x)}$, где<br>
$\sqrt{D(x)}$ - несмещенная оценка дисперсии, выполненная <font color = red>по выборке</font><br>
4) Испытания выполнены с одинаковой точностью => ГС распределена нормально<br>
5) Из 4) =>  от $\alpha$ переходим к $\frac{\alpha} { 2}$<br>
6) $X_в = \dfrac{1}{n} \cdot \sum_{i=1}^{n}X_i$

In [3]:
x = [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]
Xv = sum(x) / len(x)
print(f'Xв ={Xv: .4f}')

Xв = 6.5900


7) $s=\sqrt{\frac{1}{n-1}\cdot \sum_{i=1}^{n}(X_i-X_в)^2}$

In [4]:
s = math.sqrt(sum(map(lambda x: (x - Xv) ** 2, x)) / (10 - 1))

print(f's = {s: .4f}')

s =  0.4508


8) $K_{\frac{\alpha}{2}}=Z_{\frac{\alpha}{2}}$(из таблицы <font color = red>Стьюдента</font>)<br>
находим вероятность $p=1-\frac{\alpha}{2}$ (<font color = red>для $n-1$</font>)<br>
$Z_{\frac{\alpha}{2}} = K_{\frac{\alpha}{2}} = 2.262$


In [5]:
print(f'[{6.59 - 2.262 * (s / math.sqrt(10)): .4f}; {6.59 + 2.262 * (s / math.sqrt(10)): .4f}]')

[ 6.2675;  6.9125]


$[Xв - K \cdot Ст.ош.ср; Xв - K \cdot Ст.ош.ср]=[ 6.2675;  6.9125]$<br>

Вывод: $[ 6.2675;  6.9125]$ этот интервал покрывает истенное значение $X$ с доверительной вероятностью $95$%.

__3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.__

Дано:<br>
$\mu_0=17$<br>
$\mu=17.5$<br>
$n=100$<br>
$D(x)=4$<br>
$\alpha=5$%

In [6]:
m0 = 17
m = 17.5
n = 100
# D(x)=4
sigma = 2
alpha = 0.05

1)<br> $H_0 : \mu=\mu_0$<br>
$H_1 : \mu > \mu_0$<br>
Если $K_{расч.} > K_{табл.}$, то отвергаем $H_0$, принимаем $H_1$.<br>
2) Известна $D(x)$ из ГС => можем найти $\sigma$ из ГС<br>
$\sigma = \sqrt{D(x)} = 2$<br>
3) Из 2) => Критерий оцнки $K$ равняется критерию <font color = red>Фишера $Z$</font> и находится по таблице <font color = red>Лапласа</font> для <font color = red>$x>0$</font>.
4) Анализируем гипотезы => $K_{расч.}=Z_{расч.}=\frac{\mu-\mu_0}{\frac{\sigma}{\sqrt{n}}}$

In [7]:
print(f'Zрасч. = {(m - m0) * math.sqrt(n) / sigma}')

Zрасч. = 2.5


$K_{табл.}=Z_{табл.}=Z(p=1-\alpha)=Z(p=0.95)=1.65$

Вывод: $K_{расч.} > K_{табл.}$ => отвергаем $H_0$, принимаем $H_1$.

__4. Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:<br>
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.<br>
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?__

In [8]:
x = [202, 203, 199, 197, 195, 201, 200, 204, 194, 190]

alpha = 0.01
n = 10
m0 = 200
m = sum(x) / len(x)
print(f'm = {m}')

m = 198.5


Дано:<br>
$\alpha=1$%<br>
$n=10$<br>
$\mu_0=200$<br>
$\mu = 198.5$

1)<br>
$H_0 : \mu=\mu_0$<br>
$H_1 : \mu \not= \mu_0$<br>
Неизвестно $\sigma$ => не было анализа ГС<br>
2) Из 1) => за критерий оценки $K$ возьмем критерий <font color = red>Стьюдента</font> из таблицы <font color = red>Стьюдента</font><br>
3) Из 1) => <font color = red>$Ст.ош.ср. = \frac{s}{\sqrt{n}}$</font>,<br>
где $s$ - среднеквадратичное отклонение от несмещенной оценки дисперсии <font color = red>выборки</font>.<br>
$s=\sqrt{\frac{1}{n-1}\cdot \sum_{i=1}^{n}(X_i-X_в)^2}$

In [9]:
s = math.sqrt(sum(map(lambda x: (x - m) ** 2, x)) / (n - 1))

print(f's = {s: .4f}')

s =  4.4535


4) $K_{расч.}=Z_{расч.}=\frac{|\mu-\mu_0|}{\frac{s}{\sqrt{n}}}$

In [10]:
print(f'Kрасч. = {abs(m - m0) * math.sqrt(n) /s}')

Kрасч. = 1.0651074037450896


$K_{табл.}($<font color = red>для $n-1$</font>$)=Z_{табл.}=Z(p=1-\alpha)=Z(p=0.99)=2.821$

Вывод: $K_{расч.}<K_{табл.}$ => принимаем гипотезу $H_0$ 